src/ProtoRoute/Message.hs

```diff
@@ -14,22 +14,35 @@ module ProtoRoute.Message
 import Data.List (intercalate)
 import Data.Text (Text)
 
-newtype MsgName   = MN { unMsgName   :: String } deriving newtype (Show)
-newtype FieldName = FN { unFieldName :: String } deriving newtype (Show)
+newtype MsgName   = MN { unMsgName   :: String } deriving (Show, Eq)
+newtype FieldName = FN { unFieldName :: String } deriving (Show, Eq)
 
-data TValue a   = Req a | Opt (Maybe a) | Rep [a] | Error deriving (Show)
+data TValue a   = Req a | Opt (Maybe a) | Rep [a] | Error deriving (Show, Eq)
 data FieldValue = FText (TValue Text) | FInt (TValue Int) |
-                  FMsg (TValue Message) deriving (Show)
+                  FMsg (TValue Message) deriving (Show, Eq)
+
+data OnlyFType = Sstr | Number | Msg deriving (Show, Eq)
+data OnlyFRule = TReq | TOpt | TRep deriving (Show, Eq)
```

Comment:
Let's do some renaming. I think it's better to stick to some naming scheme. Like this one:

```haskell
data SchemaType = SText | SInt | SMsg
data SchemaRule = SReq | SOpt | SRep
```